sdk/pinocchio/src/account_info.rs

```diff
@@ -452,7 +513,9 @@ impl AccountInfo {
     pub fn close(&self) -> ProgramResult {
         // make sure the account is not borrowed since we are about to
         // resize the data to zero
-        self.check_borrow_mut_data()?;
+        if self.check_borrowed_state(BorrowState::Borrowed) {
+            return Err(ProgramError::AccountBorrowFailed);
+        }
```

Comment:
I see the bugfix here :wink:

Reply:
Yep, caught this one on the refactoring.